Api/Modpacks.cs

```diff
@@ -381,287 +171,171 @@ static class Modpacks {
-                    if (modpackConfig == null) {
-                        Utility.ShowMsg("Could not unpatch '" + modpackname + "' because the modpack's configuration is corrupted or missing." +
-                            "\r\nPlease restore from backups using the Backups tab or verify integrity of game files on Steam.", "Error");
-                        return 2;
-                    }
-                    List<modpackEntry> restored = new List<modpackEntry>(); // track restored files in case of failure mid unpatch
-                    foreach (modpackEntry entry in modpackConfig.entries) {
-                        if (String.IsNullOrEmpty(entry.type) || entry.type == "replace" || entry.type == "patch") { // assume replace type entry if null
-                            if (!Backups.RestoreBak(ExpandPath(entry.dest))) {
-                                // repatch restored mod files
-                                foreach (modpackEntry e in restored) {
-                                    int r = PatchFile(archive, e);
-                                    if (r == 2 || r == 3) {
-                                        Utility.ShowMsg("Critical error encountered while unpatching '" + modpackname + "'." +
-                                            "\r\nYou may need to verify your game files on steam or reinstall.", "Error");
-                                    }
-                                }
-                                return 2;
-                            }
-                        } else if (entry.type == "create") {
-                            if (!Utility.DeleteFile(ExpandPath(entry.dest))) {
-                                Utility.ShowMsg("Could not delete the file '" + ExpandPath(entry.dest) + "'. This may affect your game. " +
-                                    "if you encounter issue please delete this file manually.", "Warning");
-                            }
-                        } else {
-                            Utility.ShowMsg("Unknown modfile type in modpack config.\r\nCould not install the '" + modpackname + "' modpack.", "Error");
-                        }
-                        restored.Add(entry);
-                    }
-                }
-            } catch (FileNotFoundException) {
-                Utility.ShowMsg("Could not unpatch '" + modpackname + "'. Could not find the modpack file to read the config from.", "Error");
-                return 2;
-            } catch (InvalidDataException) {
-                Utility.ShowMsg("Could not unpatch '" + modpackname + "'. The modpack file appears corrupted and the config cannot be read.", "Error");
-                return 2;
+        #endregion
+
+        #region Helper Functions
+
+        private static bool EnsureModpackFolderExists() {
+            if (!Directory.Exists(Config.Modpack_dir)) {
+                _ = Directory.CreateDirectory(Config.Modpack_dir);
```

Comment:
Just curious, what's the use of assigning to an unused variable here?

Reply:
None--that change was unintentional. Removing...

I'm a little mystified as to where that came from.